### ISSUE TITLE
Redis keys not expiring

### ISSUE THREAD

**Issue by chrisjeon** [redis-store/redis-rails#19]
Hi,

We're using redis-rails with rack-attack to ip throttle cache stores.  In our production.rb and development.rb, we have defined the redis rails configuration as following:

* development.rb
config.cache_store = :redis_store, {
  host: 'localhost',
  port: 6379,
  expires_in: 2.minutes
}

It seems that the keys that are stored are not being expired after 2 minutes.  The way I'm keeping track of the keys is by opening up rails console and defining the redis object with

redis = Redis.new({ host: "localhost", port: "6379", namespace: "cache", expires_in: 2.minutes })

and then

redis.keys.count to get the total count.  Despite waiting more than 2 minutes, the count of keys is not going down.  Is my configuration incorrect?  Why aren't the keys expiring?

**Comment by rthbound** [comment]
@chrisjeon I've set up a vanilla app that uses the line from development.rb you gave.

I then cached a random string in a controller action using `Rails.cache.fetch(:bingo) { SecureRandom.hex(12) }`. I monitored the key's TTL using redis-cli and issuing `TTL bingo` over and over. The TTL was decreasing from 120s to 0s, so the `expires_in: 2.minutes` from my development.rb file must've been respected.

Some general questions: Did you cache the keys your watching through `Rails.cache#fetch` or something else? Are you sure those keys you're counting have TTLs (are you sure redis-cli wouldn't return "-1" if you issued `TTL yourkey`)?
